File: backend/app/models/event.py
from datetime import datetime
from sqlalchemy import Column, String, Integer, DateTime, ForeignKey, JSON, Boolean
from sqlalchemy.orm import relationship
from geoalchemy2 import Geography

from .base import Base

class CulturalEvent(Base):
    __tablename__ = "cultural_events"

    id = Column(String, primary_key=True)
    title = Column(String, nullable=False)
    description = Column(String)
    event_type = Column(String, nullable=False)
    date = Column(DateTime, nullable=False)
    location = Column(Geography(geometry_type='POINT', srid=4326))
    image_url = Column(String)
    created_by = Column(String, ForeignKey("users.id"))
    attendees_count = Column(Integer, default=0)
    max_attendees = Column(Integer)
    cultural_significance = Column(String)
    traditions = Column(JSON)
    required_dress_code = Column(String)
    food_preferences = Column(JSON)
    is_private = Column(Boolean, default=False)
    created_at = Column(DateTime, default=datetime.utcnow)

    # Relationships
    attendees = relationship("EventAttendee", back_populates="event")
    comments = relationship("EventComment", back_populates="event")
    photos = relationship("EventPhoto", back_populates="event")

class EventAttendee(Base):
    __tablename__ = "event_attendees"

    id = Column(String, primary_key=True)
    event_id = Column(String, ForeignKey("cultural_events.id"))
    user_id = Column(String, ForeignKey("users.id"))
    status = Column(String, nullable=False)  # going, maybe, not_going
    created_at = Column(DateTime, default=datetime.utcnow)

    # Relationships
    event = relationship("CulturalEvent", back_populates="attendees")

class EventComment(Base):
    __tablename__ = "event_comments"

    id = Column(String, primary_key=True)
    event_id = Column(String, ForeignKey("cultural_events.id"))
    user_id = Column(String, ForeignKey("users.id"))
    text = Column(String, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow)

    # Relationships
    event = relationship("CulturalEvent", back_populates="comments")

class EventPhoto(Base):
    __tablename__ = "event_photos"

    id = Column(String, primary_key=True)
    event_id = Column(String, ForeignKey("cultural_events.id"))
    user_id = Column(String, ForeignKey("users.id"))
    photo_url = Column(String, nullable=False)
    caption = Column(String)
    created_at = Column(DateTime, default=datetime.utcnow)

    # Relationships
    event = relationship("CulturalEvent", back_populates="photos")

File: pendo_mobile/lib/features/events/domain/entities/event.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'event.freezed.dart';
part 'event.g.dart';

@freezed
class CulturalEventModel with _$CulturalEventModel {
  const factory CulturalEventModel({
    required String id,
    required String title,
    required String description,
    @JsonKey(name: 'event_type') required String eventType,
    required DateTime date,
    required String location,
    @JsonKey(name: 'image_url') String? imageUrl,
    @JsonKey(name: 'created_by') required String createdBy,
    @JsonKey(name: 'attendees_count') required int attendeesCount,
    @JsonKey(name: 'max_attendees') int? maxAttendees,
    @JsonKey(name: 'cultural_significance') String? culturalSignificance,
    @JsonKey(name: 'traditions') List<String>? traditions,
    @JsonKey(name: 'required_dress_code') String? requiredDressCode,
    @JsonKey(name: 'food_preferences') List<String>? foodPreferences,
    @JsonKey(name: 'is_private') required bool isPrivate,
    @JsonKey(name: 'created_at') required DateTime createdAt,
  }) = _CulturalEventModel;

  factory CulturalEventModel.fromJson(Map<String, dynamic> json) =>
      _$CulturalEventModelFromJson(json);
}

@freezed
class EventAttendeeModel with _$EventAttendeeModel {
  const factory EventAttendeeModel({
    required String id,
    @JsonKey(name: 'event_id') required String eventId,
    @JsonKey(name: 'user_id') required String userId,
    required String status, // going, maybe, not_going
    @JsonKey(name: 'created_at') required DateTime createdAt,
  }) = _EventAttendeeModel;

  factory EventAttendeeModel.fromJson(Map<String, dynamic> json) =>
      _$EventAttendeeModelFromJson(json);
}

@freezed
class EventCommentModel with _$EventCommentModel {
  const factory EventCommentModel({
    required String id,
    @JsonKey(name: 'event_id') required String eventId,
    @JsonKey(name: 'user_id') required String userId,
    required String text,
    @JsonKey(name: 'created_at') required DateTime createdAt,
  }) = _EventCommentModel;

  factory EventCommentModel.fromJson(Map<String, dynamic> json) =>
      _$EventCommentModelFromJson(json);
}

@freezed
class EventPhotoModel with _$EventPhotoModel {
  const factory EventPhotoModel({
    required String id,
    @JsonKey(name: 'event_id') required String eventId,
    @JsonKey(name: 'user_id') required String userId,
    @JsonKey(name: 'photo_url') required String photoUrl,
    String? caption,
    @JsonKey(name: 'created_at') required DateTime createdAt,
  }) = _EventPhotoModel;

  factory EventPhotoModel.fromJson(Map<String, dynamic> json) =>
      _$EventPhotoModelFromJson(json);
}
